pypesto/sampling/diagnostics.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from ..result import Result
+from .geweke_test import burn_in_by_sequential_geweke
+from .auto_correlation import auto_correlation
+
+
+def geweke_test(result: Result,
+                zscore: float = 2.):
+    """
+    Calculates the burn-in of MCMC chains.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    zscore:
+        The Geweke test threshold. Default 2.
+
+    Returns
+    -------
+    burn_in:
+        Iteration where the first and the last fraction of the chain
+        do not differ significantly regarding Geweke test -> Burn-In
+
+    """
+    # Get parameter samples as numpy arrays
+    chain = np.array(result.sample_result['trace_x'][0])
```

Comment:
better use `sample_result.trace_x` (similar below) as this will enable automatic code checking